launch/launch/frontend/type_utils.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module which implements get_typed_value function."""
+
+from typing import Any
+from typing import List
+from typing import Text
+from typing import Tuple
+from typing import Union
+
+from .entity import Entity
+
+__ScalarTypesTuple = (
+    int, float, bool, str
+)
+
+__TypesForGuessTuple = (
+    int, float, bool, list, str
+)
+
+
+def check_is_list(data_type: Any) -> bool:
+    """Check if `data_type` was origined from `typing.List`."""
+    return hasattr(data_type, '__origin__') and \
+        data_type.__origin__ in (list, List)  # On Linux/Mac is List, on Windows is list.
+
+
+def check_is_union(data_type: Any) -> bool:
+    """Check if `data_type` was origined from `typing.Union`."""
+    return hasattr(data_type, '__origin__') and \
+        data_type.__origin__ is Union
+
+
+def check_is_list_entity(data_type: Any) -> bool:
+    """
+    Return `True` if `data_type` is `List[launch.frontend.Entity]`.
+
+    It returns also `True` the member type is a subclass of `launch.frontend.Entity`.
+    """
+    return check_is_list(data_type) and \
+        issubclass(data_type.__args__[0], Entity)
+
+
+def get_tuple_of_types(data_type: Any) -> Tuple:
+    """Convert typing.Union to tuple of types. If not, return `(data_type,)`."""
```

Comment:
@ivanpauno consider
```suggestion
    """
    Normalize `data_type` to a tuple of types. 
    
    If `data_type` is based on a `typing.Union`, return union types. 
    Otherwise, return a `(data_type,)` tuple.
    """
```